### DISCUSSION TITLE
How to use machine learning potentials to track the change in species numbers during a reaction process.

### DISCUSSION THREAD

**Discussion by jzh-w** [deepmodeling/reacnetgenerator#2321] [Q&A]
How can a machine learning potential trained with deepmd-kit be used to track the change in the number of intermediates during a reaction process, similar to how reactive force fields like ReaxFF do?

**Comment by njzjz** [comment]
The trajectory can be processed by ReacNetGenerator, https://github.com/deepmodeling/reacnetgenerator

**Reply by jzh-w** [reply]
Thank you for your response! However, I encountered a situation during usage that differs from the example in the tutorial. Specifically, the process gets stuck at the first step and takes a long time to run. Initially, I suspected it was due to the large file size and made a series of attempts to address it. However, after reviewing the example in the tutorial, I found that the total runtime was only 2.87 seconds. Therefore, I am now considering whether it might be due to my computer's configuration, but I still hope to get an accurate answer. I would greatly appreciate it if you could reply to this message. Thank you very much!
![命令照片](https://github.com/user-attachments/assets/3c4aa9b3-5951-45c8-831b-235923609001)

**Reply by jzh-w** [reply]
Hello! I tried using a better device to run the operation, but I got the same result. Now I suspect that ReacNetGenerator might not have been installed successfully, as the version displayed in the image is "Unknown". Additionally, when I run `reacnetgenerator --version`, it shows "ReacNetGenerator vUnknown". However, when I use `pip show reacnetgenerator`, it shows the result as `Name: reacnetgenerator Version: 1.6.14`. Therefore, I would like to ask for your advice on what might be causing this issue.
![5ac927f583e7643e753bff2fb7313d1](https://github.com/user-attachments/assets/7195b35e-7214-4db8-b80c-54318b9feb9c)
![0d9407d2f381508cb5dd918ed55ec50](https://github.com/user-attachments/assets/441069dc-bb2f-49c1-8ddd-fd9e4c6497ad)

**Reply by njzjz** [reply]
How large is your file? The file size in the example is not so much.

**Reply by jzh-w** [reply]
The trajectory file for the combustion process is 2.13 GB, which is indeed very large. So, it seems that my ReacNetGenerator installation should be fine, right? Should I still consider the version issue? When you wrote the paper **"Complex reaction processes in combustion unraveled by neural network-based molecular dynamics simulation"**, approximately how long did it take to process the trajectory file? I would like to use it as a reference. Are there any good methods to speed up this process?

**Reply by njzjz** [reply]
2.13GB is not very large. Did you use a supercomputer node with many CPU cores?

**Reply by jzh-w** [reply]
I am using the server configured by my research group, not a supercomputer. Additionally, when using my own machine, I noticed that the CPU cores were not fully utilized.

**Reply by njzjz** [reply]
I suggest using a machine with multiple cores and enough memory and waiting for a night.

**Reply by jzh-w** [reply]
Okay, thank you! I will follow your suggestion and try again. Since I noticed during yesterday's run that the CPU cores were not fully utilized, even though the ReacNetGenerator documentation states that `-np` defaults to using all available processes, I want to set it manually. However, I am concerned that I might input the command incorrectly. So could you please check if the command I wrote can achieve the desired effect:

```bash
reacnetgenerator -i relax.lammpstrj -a B O -np 32 --type dump --nohmm
```
Once again, I would like to express my gratitude to you!

**Reply by njzjz** [reply]
> ```shell
> reacnetgenerator -i relax.lammpstrj -a B O -np 32 --type dump --nohmm
> ```

This looks good to me.

**Reply by jzh-w** [reply]
Thank you once again! Recently, I read the content you published on "Zhihu", and it left a deep impression on me. I wish you an even brighter future!Thank you!
